src/template/commands/time.rs

```diff
@@ -0,0 +1,40 @@
+use std::collections::HashSet;
+
+use crate::template::run_multi::run_multi;
+use crate::template::timings::Timings;
+use crate::template::{all_days, readme_benchmarks, Day};
+
+pub fn handle(day: Option<Day>, recreate_all: bool) {
+    let stored_timings = Timings::read_from_file();
+
+    let mut days_to_run = HashSet::new();
+
+    match day {
+        Some(day) => {
+            days_to_run.insert(day);
+        }
+        None => {
+            all_days().for_each(|day| {
+                // when the `--all` flag is not set, filter out days that are fully benched.
+                if recreate_all || !stored_timings.is_day_complete(&day) {
+                    days_to_run.insert(day);
+                }
+            });
+        }
+    };
```

Comment:
```suggestion
    let days_to_run = day.map(|day| HashSet::from([day])).unwrap_or_else(|| {
        if recreate_all {
            all_days().collect()
        } else {
            // when the `--all` flag is not set, filter out days that are fully benched.
            all_days().filter(|day| !stored_timings.is_day_complete(&day)).collect()
        }
    });
```

It is possible to simplify to construction of `days_to_run`.
With this change there is no need of mutable variable and the evaluation of `recreate_all` is moved outside the loop.